hkmc2/shared/src/main/scala/hkmc2/codegen/llir/Llir.scala

```diff
@@ -47,67 +56,42 @@ implicit object ClassInfoOrdering extends Ordering[ClassInfo] {
 
 case class ClassInfo(
   id: Int,
-  name: Str,
-  fields: Ls[Str],
+  name: Local,
+  fields: Ls[Local],
+  parents: Set[Local],
+  methods: Map[Local, Func],
 ):
-  var parents: Set[Str] = Set.empty
-  var methods: Map[Str, Func] = Map.empty
   override def hashCode: Int = id
   override def toString: String =
     s"ClassInfo($id, $name, [${fields mkString ","}], parents: ${parents mkString ","}, methods:\n${methods mkString ",\n"})"
 
   def show = toDocument.toString
   def toDocument: Document =
     given Conversion[String, Document] = raw
-    val ext = if parents.isEmpty then "" else " extends " + parents.mkString(", ")
+    val ext = if parents.isEmpty then "" else " extends " + parents.map(_.nme).mkString(", ")
     if methods.isEmpty then
-      doc"class $name(${fields.mkString(",")})$ext"
+      doc"class ${name.nme}(${fields.map(docSymWithUid).mkString(",")})$ext"
     else
-      val docFirst = doc"class $name (${fields.mkString(",")})$ext {"
+      val docFirst = doc"class ${name.nme}(${fields.map(docSymWithUid).mkString(",")})$ext {"
       val docMethods = methods.map { (_, func) => func.toDocument }.toList.mkDocument(doc" # ")
       val docLast = doc"}"
-      doc"$docFirst #{  # $docMethods #  #} $docLast"
-
-case class Name(str: Str):
-  def trySubst(map: Map[Str, Name]) = map.getOrElse(str, this)
-  override def toString: String = str
-
-object FuncRef:
-  def fromName(name: Str) = FuncRef(Right(name))
-  def fromName(name: Name) = FuncRef(Right(name.str))
-  def fromFunc(func: Func) = FuncRef(Left(func))
+      doc"$docFirst #{  # $docMethods #}  # $docLast"
 
-class FuncRef(var func: Either[Func, Str]):
-  def name: String = func.fold(_.name, x => x)
-  def expectFn: Func = func.fold(identity, x => throw Exception(s"Expected a def, but got $x"))
-  def getFunc: Opt[Func] = func.left.toOption
+class FuncRef(var func: Local):
+  def name: String = func.nme
   override def equals(o: Any): Bool = o match {
     case o: FuncRef => o.name == this.name
     case _ => false
   }
 
-object ClassRef:
-  def fromName(name: Str) = ClassRef(Right(name))
-  def fromName(name: Name) = ClassRef(Right(name.str))
-  def fromClass(cls: ClassInfo) = ClassRef(Left(cls))
-
-class ClassRef(var cls: Either[ClassInfo, Str]):
-  def name: String = cls.fold(_.name, x => x) 
-  def expectCls: ClassInfo = cls.fold(identity, x => throw Exception(s"Expected a class, but got $x"))
-  def getClass: Opt[ClassInfo] = cls.left.toOption
-  override def equals(o: Any): Bool = o match {
-    case o: ClassRef => o.name == this.name
-    case _ => false
-  }
-
 implicit object FuncOrdering extends Ordering[Func] {
   def compare(a: Func, b: Func) = a.id.compare(b.id)
 }
 
 case class Func(
   id: Int,
-  name: Str,
-  params: Ls[Name],
+  name: Local,
```

Comment:
Can this be
```suggestion
  sym: TermSymbol,
```
or
```suggestion
  sym: BlockMemberSymbol,
```
?